documentation/homework/week4/src/stores/todo_actions.js

```diff
@@ -0,0 +1,150 @@
+import { observable, action, runInAction } from 'mobx'
+
+// Setting our localStorage
+// import { locals } from '../utils'
+
+// set the default state of the items Object
+const default_item_state = {
+  description: '',
+  deadline: '',
+}
+
+async function request(url, method, inputToString) {
+  const params = {
+    "body": JSON.stringify(inputToString),
+    "headers": {
+      "Content-type": "application/json"
+    }
+  }
+
+  let options = {}
+  method = method ? method.toUpperCase() : "GET"
+  switch (method) {
+    case "POST":
+    case "PATCH":
+      options = { method, ...params, }
+      break
+    default:
+      options = { method }
+  }
+
+  // is't Okay to catch the errors here with in 
+  // -- one single line with the.json line ?
+  // OR we have to catch it on the fetch it self also ?!
+  // Because what i see here is a strange look to me
+  const preResponse = await fetch(`https://hyf-react-api.herokuapp.com/${url}`, options)
+  const response = await preResponse.json().catch(console.error)
+  return response
+}
+
+class todo_actions {
```

Comment:
This is not a good name for your store. Your store contains a lot more than just actions! It also contains observables, and can contain computed values etc. It should be called `TodoStore` in capitalised case like that.